cmd/minishift/cmd/root.go

```diff
@@ -79,12 +79,12 @@ var RootCmd = &cobra.Command{
 	Long:  `Minishift is a command-line tool that provisions and manages single-node OpenShift clusters optimized for development workflows.`,
 	PersistentPreRun: func(cmd *cobra.Command, args []string) {
 		var (
-			err          error
-			isFreshStart bool
+			err                    error
+			isAddonInstallRequired bool
 		)
 
-		if !filehelper.Exists(constants.Minipath) || filehelper.IsEmptyDir(constants.Minipath) {
-			isFreshStart = true
+		if !filehelper.Exists(constants.MakeMiniPath("addons")) {
```

Comment:
Works for me. The big issue I am having with this code, is the use of constants.MakeMiniPath("addons"). This is really a misuse of the constants and very error prone. However we are not only doing this here, but for other paths as well. I am hoping to be able to address this once we have some sort of MinishiftPaths struct which gives typed access to the various paths in $MINISHIFT_HOME. AFAIK @LalatenduMohanty is creating this struct as part of the profile feature. 

I already see this code breaking due to this. Really we should have named these variables a long time ago in the constant files, e.g. constants.AddonPath or something. Same for the other paths. Using constants.MakeMiniPath("addons") is just wrong.